plugin/web/extensions/infobox/core/components/viewer/index.tsx

```diff
@@ -0,0 +1,95 @@
+import type { Primitive, PublicSetting } from "@web/extensions/infobox/types";
+import { Collapse } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import { useEffect, useState } from "react";
+
+type Props = {
+  key: number;
+  primitive: Primitive;
+  publicSettings: PublicSetting[];
+};
+
+type PropertyItemType = {
+  key: string;
+  title: string;
+  value?: any;
+};
+
+const Viewer: React.FC<Props> = ({ primitive, publicSettings, key, ...props }) => {
+  const [propertyList, setPropertyList] = useState<PropertyItemType[]>([]);
+
+  useEffect(() => {
+    const propertyItem: PropertyItemType[] = [];
+    const publicSetting = publicSettings.find(ps => ps.type === primitive.type);
+    if (!publicSetting || publicSetting.properties.length === 0) {
+      primitive.properties.forEach(property => {
```

Comment:
Typically you should avoid forEach because it'll mutate the original. Are you doing this on purpose, or perhaps you should make the primitive as a useState and use its setter function to modify it?
If you think this is fine, please let me know as I am a little unsure when mutating the original with a forEach is actually desirable (at least inside of React).